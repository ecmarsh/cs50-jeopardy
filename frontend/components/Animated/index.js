import Ease from './_Ease';
import Fade from './_Fade';
import Grow from './_Grow';
import Loud from './_Loud';
import Shift from './_Shift';
import Slide from './_Slide';
import Tally from './_Tally';
import Tree from './_Tree';
import Unfocus from './_Unfocus';

export { Ease, Fade, Grow, Loud, Shift, Slide, Tally, Tree, Unfocus };
