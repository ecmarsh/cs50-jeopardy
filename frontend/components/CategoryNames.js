import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Mutation } from 'react-apollo';
import gql from 'graphql-tag';
import * as R from 'ramda';
import formatRound from '../lib/formatRound';
import Error from './ErrorMessage';
import Form from './styles/Form';
import Button from './styles/Button';
import { TextMuted, Divider } from './styles/Utilities';

const UPDATE_CATEGORY_MUTATION = gql`
  mutation UPDATE_CATEGORY_MUTATION($categoryId: ID!, $categoryName: String) {
    updateCategory(id: $categoryId, name: $categoryName) {
      id
      name
    }
  }
`;

class CategoryNames extends Component {
  static propTypes = {
    categories: PropTypes.array.isRequired,
  };

  state = {
    categories: [],
    called: false,
  };

  componentDidMount() {
    const { categories } = this.props;
    this.setState({ categories });
  }

  handleChange = e => {
    e.preventDefault();
    const { id, value } = e.target;
    const categories = [...this.state.categories];
    const idx = R.findIndex(R.propEq('id', id))(categories);
    categories[idx].name = value;
    categories[idx].updated = true; // flag to include in mutation
    this.setState({ categories });
  };

  handleSubmit = (e, updateCategoryMutation) => {
    // Stop default form submission
    e.preventDefault();
    // Get list of changed categories to update
    const updatedCategories = R.filter(R.has('updated'), this.state.categories);
    // Update category mutation
    const doMutation = async category => {
      const { id, name } = category;
      const res = await updateCategoryMutation({
        variables: { categoryId: id, categoryName: name },
      });
    };
    // Invoke mutation on each updated category
    R.forEach(doMutation, updatedCategories);
    // Called to flash saved message
    this.setState({ called: true });

    setTimeout(this.flashUpdatedConfirmation, 2000);
  };

  flashUpdatedConfirmation = () => {
    this.setState({ called: false });
  };

  render() {
    const { categories, called } = this.state;
    return (
      <Mutation mutation={UPDATE_CATEGORY_MUTATION}>
        {(updateCategory, { loading, error }) => {
          return (
            <Form
              data-test="category-names-form"
              className="noBg"
              onSubmit={e => this.handleSubmit(e, updateCategory)}
            >
              <h2>Category Names</h2>
              <Error error={error} />
              <fieldset disabled={loading} aria-busy={loading}>
                {categories.map((category, i) => (
                  <div key={category.id}>
                    {i === 0 && <h4>{formatRound(1)}</h4>}
                    {i === 5 && <h4>{formatRound(2)}</h4>}
                    {i === 10 && <h4>{formatRound(3)}</h4>}
                    <label
                      htmlFor={`category-${i}`}
                      style={{
                        flexDirection: 'row',
                        width: '100%',
                      }}
                    >
                      <TextMuted
                        style={{ textAlign: 'center', flexBasis: '25%' }}
                      >
                        CATEGORY {i + 1}:
                      </TextMuted>
                      <input
                        type="text"
                        name={`category-${i}`}
                        id={category.id}
                        value={category.name}
                        onChange={e => this.handleChange(e)}
                        style={{
                          flexBasis: '60%',
                          width: 'auto',
                        }}
                      />
                    </label>
                  </div>
                ))}
                <div style={{ margin: '0 auto' }}>
                  <TextMuted
                    data-test="category-names-save-confirmation"
                    hidden={!called}
                  >
                    Category Names Saved!
                  </TextMuted>
                  <Button
                    primary
                    type="submit"
                    style={{ margin: '1rem auto 2rem' }}
                  >
                    Save Category Names
                  </Button>
                  <Divider />
                  {this.props.children}
                </div>
              </fieldset>
            </Form>
          );
        }}
      </Mutation>
    );
  }
}

export default CategoryNames;
export { UPDATE_CATEGORY_MUTATION };
