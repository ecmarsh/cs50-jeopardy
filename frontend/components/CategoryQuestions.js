import React from 'react';
import { Query } from 'react-apollo';
import gql from 'graphql-tag';
import AddToCategory from './CategoryQuestionAdd';
import { categoriesPerPage } from '../config';

const CATEGORY_QUESTIONS_QUERY = gql`
  query CATEGORY_QUESTIONS_QUERY($categoryId: ID) {
    categoryQuestions(where: { category: { id: $categoryId } }) {
      id
      difficulty
      isDouble
      answered
      question {
        id
        question
        answer
      }
      category {
        id
        name
      }
    }
  }
`;

const CategoryQuestions = ({ gameName, categoryId, round }) => (
  <Query query={CATEGORY_QUESTIONS_QUERY} variables={{ categoryId }}>
    {({ data, error }) => {
      if (error) return <p>Error...</p>;
      if (data.categoryQuestions)
        return (
          <>
            {data.categoryQuestions.map((categoryQuestion, index) => (
              <AddToCategory
                key={categoryQuestion.id}
                gameName={gameName}
                categoryId={categoryId}
                categoryName={categoryQuestion.category.name}
                categoryQuestionId={categoryQuestion.id}
                difficulty={categoryQuestion.difficulty}
                question={categoryQuestion.question}
                loaded={categoryQuestion.question !== null}
                skip={categoriesPerPage * (round - 1)}
                cacheCategoryIndex={index}
              />
            ))}
          </>
        );
    }}
  </Query>
);

export default CategoryQuestions;
export { CATEGORY_QUESTIONS_QUERY };
