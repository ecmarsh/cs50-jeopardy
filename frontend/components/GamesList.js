import React from 'react';
import PropTypes from 'prop-types';
import { Query } from 'react-apollo';
import gql from 'graphql-tag';
import Error from './ErrorMessage';

const GAMES_QUERY = gql`
  query GAMES_QUERY {
    games {
      id
      name
      isPublic
      user {
        id
      }
    }
  }
`;

const GameOptions = ({ htmlName, handler }) => (
  <Query query={GAMES_QUERY}>
    {({ data: { games }, loading, error }) => {
      if (loading) return 'Loading...';
      if (error) return <Error error={error} />;
      return (
        <select
          name={htmlName}
          id={htmlName}
          autoFocus
          required
          onChange={handler}
          defaultValue="Select Game"
        >
          <option disabled>Select Game</option>
          {games.map(game => (
            <option key={game.id} value={game.name}>
              {game.name}
            </option>
          ))}
        </select>
      );
    }}
  </Query>
);

GameOptions.propTypes = {
  htmlName: PropTypes.string.isRequired,
  handler: PropTypes.func.isRequired,
};

export default GameOptions;
export { GAMES_QUERY };
