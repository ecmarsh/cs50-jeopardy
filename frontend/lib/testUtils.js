import casual from 'casual';
import mappingArray from './mappingArray';

// seed it for repeatable random sequence
casual.seed(123);

const fakeQuestion = () => ({
  id: 'abc123',
  question: 'Time complexity of merge sort?',
  answer: 'n log n',
  user: {
    id: casual.uuid,
    name: 'User Name',
    __typename: 'User',
  },
  createdAt: '2019-02 - 26T19: 24: 16.000Z',
  game: {
    id: 'game123',
    name: 'GameForGeekz',
    user: {
      id: casual.uuid,
      __typename: 'User',
    },
    isPublic: casual.coin_flip,
    __typename: 'Game',
  },
  categoryQuestions: [
    {
      id: 'catQuest123',
      category: { id: 'cat123', __typename: 'Category' },
      __typename: '__CategoryQuestion',
    },
  ],
  __typename: 'Question',
});

const fakeUser = () => ({
  __typename: 'User',
  id: 'userABC123',
  name: casual.name,
  email: casual.email,
  permissions: ['USER'],
});

const fakeAdmin = () => ({
  __typename: 'User',
  id: 'adminABC123',
  name: casual.name,
  email: casual.email,
  permission: ['ADMIN', 'USER'],
  games: [],
});

const fakeCategoryQuestion = (id = 'catQuest123', difficulty = 1) => ({
  id,
  difficulty,
  isDouble: false,
  answered: casual.coin_flip,
  question: fakeQuestion(),
  category: { id: 'cat123', name: casual.word, __typename: 'Category' },
  game: { id: 'g@m3', name: 'Test Game', __typename: 'Game' },
  __typename: 'CategoryQuestion',
});

const fillCategoryQuestions = (n = 5) =>
  mappingArray(n).map(i => fakeCategoryQuestion(`catQuest-${i}`, i));

const fakeCategory = (id = 'cat123') => ({
  id,
  name: casual.word,
  game: { id: 'g@m3', name: 'Test Game' },
  categoryQuestions: fillCategoryQuestions(),
  __typename: 'Category',
});

const fillCategories = (n = 11) =>
  mappingArray(n).map(i => fakeCategory(`cat-${i}`));

const fakeTeam = id => ({
  __typename: 'Team',
  id: `t${id}`,
  name: `Team${id}`,
  score: 1000,
  game: { id: 'g@m3', name: 'Test Game' },
});

const fakeGameConfig = () => ({
  id: 'config123',
  hasDoubleJeopardy: false,
  hasRoundTimer: false,
  roundTime: 0,
  finalTime: 0,
  __typename: 'GameConfig',
});

const fakeGame = () => ({
  __typename: 'Game',
  id: 'g@m3',
  name: 'Test Game',
  createdAt: '2019-03 - 06T19: 12: 16.000Z',
  isPublic: casual.coin_flip,
  user: fakeUser(),
  teams: [fakeTeam(1), fakeTeam(2)],
  categories: fillCategories(),
  questions: [fakeQuestion()],
  config: fakeGameConfig(),
});

const fakeStudyGame = (id = 'SG123') => ({
  __typename: 'StudyGame',
  id,
  answeredQuestions: [fakeCategoryQuestion(), fakeCategoryQuestion()],
  score: 1000,
  game: fakeGame(),
  user: fakeUser(),
  createdAt: '2019-04 - 25T19: 10: 16.000Z',
});

const fakeGameSummary = () => ({
  questionsConnection: {
    aggregate: {
      count: casual.double + 1,
      __typename: 'aggregate',
    },
    __typename: 'questionsConnection',
  },
  categoryQuestionsConnection: {
    aggregate: {
      count: casual.double,
      __typename: 'aggregate',
    },
    __typename: 'categoryQuestionsConnection',
  },
});

// Fake LocalStorage
class LocalStorageMock {
  constructor() {
    this.store = {};
  }

  clear() {
    this.store = {};
  }

  getItem(key) {
    return this.store[key] || null;
  }

  setItem(key, value) {
    this.store[key] = value.toString();
  }

  removeItem(key) {
    delete this.store[key];
  }
}

export {
  LocalStorageMock,
  fakeQuestion,
  fakeUser,
  fakeAdmin,
  fakeCategory,
  fakeCategoryQuestion,
  fakeTeam,
  fakeGame,
  fakeStudyGame,
  fakeGameSummary,
  fakeGameConfig,
};
