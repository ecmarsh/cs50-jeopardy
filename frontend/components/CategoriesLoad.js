import React, { Component } from 'react';
import { Query } from 'react-apollo';
import gql from 'graphql-tag';
import { categoriesPerPage } from '../config';
import CategoryQuestions from './CategoryQuestions';
import { Heading } from './Views';
import Pagination from './Pagination';
import Error from './ErrorMessage';
import Main from './styles/Main';
import { Divider } from './styles/Utilities';

const CATEGORIES_QUERY = gql`
  query CATEGORIES_QUERY($gameName: String, $first: Int, $skip: Int) {
    categories(
      where: { game: { name: $gameName } }
      first: $first
      skip: $skip
    ) {
      id
      name
    }
  }
`;

class LoadCategories extends Component {
  state = {};

  handleChange = e => {
    this.setState({ [e.target.name]: e.target.value });
  };

  render() {
    const { game, round, headingProps } = this.props;
    return (
      <Query
        query={CATEGORIES_QUERY}
        variables={{
          gameName: game,
          first: categoriesPerPage,
          skip: categoriesPerPage * (round - 1),
        }}
      >
        {({ data, loading, error }) => {
          if (loading) return <p>Loading...</p>;
          if (error) return <Error error={error} />;
          if (!data.categories) {
            return <Heading game={game} {...headingProps} />;
          }
          return (
            <Main>
              <Heading game={game} {...headingProps} />
              <Divider style={{ margin: '0 auto' }} />
              <Pagination gameName={game} round={round} />
              <div className="canvas">
                {data.categories.map(category => (
                  <div
                    key={'category' + category.id}
                    className="column-wrapper"
                  >
                    <ul>
                      <li>{category.name}</li>
                      <CategoryQuestions
                        categoryId={category.id}
                        gameName={game}
                        round={round}
                      />
                    </ul>
                  </div>
                ))}
              </div>
            </Main>
          );
        }}
      </Query>
    );
  }
}

export default LoadCategories;
export { CATEGORIES_QUERY };
