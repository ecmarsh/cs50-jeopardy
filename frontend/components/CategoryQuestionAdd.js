import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Mutation } from 'react-apollo';
import gql from 'graphql-tag';
import DropTarget from './styles/DropTarget';
import Error from './ErrorMessage';
import RemoveFromCategory from './CategoryQuestionRemove';
import { GAME_QUESTIONS_QUERY } from './QuestionsList';
import { CATEGORY_QUESTIONS_QUERY } from './CategoryQuestions';
import { GAME_SUMMARY_QUERY } from './GameSummary';
import * as R from 'ramda';

const ADD_TO_CATEGORY_MUTATION = gql`
  mutation ADD_TO_CATEGORY_MUTATION($id: ID!, $questionId: ID!) {
    updateCategoryQuestion(id: $id, questionId: $questionId) {
      id
      question {
        categoryQuestions {
          id
        }
      }
    }
  }
`;

class AddToCategory extends Component {
  static propTypes = {
    loaded: PropTypes.bool,
    question: PropTypes.object,
    categoryQuestionId: PropTypes.string.isRequired,
    categoryId: PropTypes.string.isRequired,
    difficulty: PropTypes.number.isRequired,
    categoryName: PropTypes.string.isRequired,
    gameName: PropTypes.string.isRequired,
  };

  state = {
    questionId: String(),
    question: String(),
  };

  componentDidMount = () => {
    if (this.props.question) {
      const { id, question } = this.props.question;
      this.setState({
        questionId: id,
        question,
      });
    }
  };

  handleDragOver = e => {
    e.preventDefault();
    e.persist(); // don't pool the events
    e.target.style.transform = 'scale(1.2)';
    e.dataTransfer.dropEffect = 'move';
  };

  handleDragLeave = e => {
    e.target.style.transform = 'scale(1)';
  };

  handleDrop = async (e, categoryMutation) => {
    e.persist();
    e.target.style.transform = 'scale(1)';
    e.dataTransfer.dropEffect = 'move';

    const { categoryId, categoryQuestionId, difficulty, gameName } = this.props,
      questionId = e.dataTransfer.getData('id'),
      question = e.dataTransfer.getData('question'),
      answer = e.dataTransfer.getData('answer');

    this.setState({ questionId, question });

    const res = await categoryMutation({
      variables: { id: categoryQuestionId, questionId },
      update: (cache, { data: { updateCategoryQuestion } }) => {
        /* Update category question with added question */
        const categoryQuestionsQuery = {
          query: CATEGORY_QUESTIONS_QUERY,
          variables: { categoryId },
        };
        // Read the data
        const { categoryQuestions } = cache.readQuery(categoryQuestionsQuery);
        // Add the question
        categoryQuestions[difficulty - 1].question = {
          answer,
          id: questionId,
          question,
          __typename: 'Question',
        };
        // Write back the cache
        cache.writeQuery({
          data: { categoryQuestions },
          ...categoryQuestionsQuery,
        });

        /* Update question to know it's in a category now */
        const gameQuestionsQuery = {
          query: GAME_QUESTIONS_QUERY,
          variables: { gameName },
        };
        // Read data to questions variable
        const { questions } = cache.readQuery(gameQuestionsQuery);
        // Add new question into matching question's category list
        const idx = R.findIndex(R.propEq('id', questionId), questions);
        const questionToAdd = {
          id: updateCategoryQuestion.id,
          __typename: 'CategoryQuestion',
          category: {
            id: categoryId,
            __typename: 'Category',
          },
        };
        questions[idx].categoryQuestions.push(questionToAdd);
        // Write back cache with updates
        gameQuestionsQuery.data = { questions };
        //console.log(gameQuestionsQuery);
        cache.writeQuery(gameQuestionsQuery);
      },
    });
  };

  render() {
    const {
      loaded,
      question,
      categoryQuestionId,
      categoryId,
      difficulty,
      categoryName,
      gameName,
    } = this.props;

    return (
      <Mutation
        mutation={ADD_TO_CATEGORY_MUTATION}
        refetchQueries={[
          {
            query: GAME_SUMMARY_QUERY,
            variables: { gameName },
          },
        ]}
      >
        {(updateCategoryQuestion, { loading, error }) => {
          if (loading) return <DropTarget>Adding..</DropTarget>;
          if (error) return <Error error={error} />;
          return (
            <DropTarget
              id={`${categoryName}-${categoryId}`}
              onDragOver={!loaded ? this.handleDragOver : undefined}
              onDragLeave={!loaded ? this.handleDragLeave : undefined}
              onDrop={e => this.handleDrop(e, updateCategoryQuestion)}
              className={loaded ? '--isLoaded' : '--isNotLoaded'}
            >
              {loaded ? <div>{question.question}</div> : difficulty}
              {loaded && (
                <RemoveFromCategory
                  categoryQuestionId={categoryQuestionId}
                  categoryId={categoryId}
                  gameName={gameName}
                  difficulty={difficulty}
                  questionId={this.state.questionId}
                />
              )}
            </DropTarget>
          );
        }}
      </Mutation>
    );
  }
}

export default AddToCategory;
export { ADD_TO_CATEGORY_MUTATION };
