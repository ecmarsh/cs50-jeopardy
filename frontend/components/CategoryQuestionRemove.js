import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Mutation } from 'react-apollo';
import gql from 'graphql-tag';
import Button from './styles/Button';
import Error from './ErrorMessage';
import { CATEGORY_QUESTIONS_QUERY } from './CategoryQuestions';
import { GAME_QUESTIONS_QUERY } from './QuestionsList';
import { GAME_SUMMARY_QUERY } from './GameSummary';
import { TextLarger } from './styles/Utilities';
import * as R from 'ramda';

const REMOVE_FROM_CATEGORY_MUTATION = gql`
  mutation REMOVE_FROM_CATEGORY_MUTATION($id: ID!) {
    removeFromCategory(id: $id) {
      id
    }
  }
`;

class RemoveFromCategory extends Component {
  static propTypes = {
    children: PropTypes.string,
    categoryQuestionId: PropTypes.string.isRequired,
  };
  static defaultProps = {
    children: `âœ•`,
  };

  handleClick = async (e, removeFromCategoryMutation) => {
    e.preventDefault();
    const {
      questionId,
      categoryQuestionId,
      categoryId,
      difficulty,
      gameName,
    } = this.props;

    const res = await removeFromCategoryMutation({
      // Mutation variables
      variables: {
        id: categoryQuestionId,
      },
      // Update local data
      update: (cache, { data: { removeFromCategory } }) => {
        // Update the question so it knows it's not in a category
        const { questions } = cache.readQuery({
          query: GAME_QUESTIONS_QUERY,
          variables: { gameName },
        });
        const questionIndex = questions.findIndex(
          question => question.id === questionId
        );

        const idx = R.findIndex(R.propEq('id', questionId));

        // Filter the category question out
        questions[questionIndex].categoryQuestions = questions[
          questionIndex
        ].categoryQuestions.filter(
          categoryQuestion => categoryQuestion.id !== removeFromCategory.id
        );

        // Write back the questions data
        cache.writeQuery({
          query: GAME_QUESTIONS_QUERY,
          variables: { gameName },
          data: { questions },
        });
        // Read, update, writeback the loaded questions cache
        const { categoryQuestions } = cache.readQuery({
          query: CATEGORY_QUESTIONS_QUERY,
          variables: { categoryId },
        });
        // Remove the question from the category
        categoryQuestions[difficulty - 1].question = null;
        // Write back the cache
        cache.writeQuery({
          query: CATEGORY_QUESTIONS_QUERY,
          variables: { categoryId },
          data: { categoryQuestions },
        });
      },
      optimisticResponse: {
        __typename: 'Mutation',
        removeFromCategory: {
          id: categoryQuestionId,
          __typename: 'CategoryQuestion',
        },
      },
    });
  };

  render() {
    const { gameName } = this.props;
    return (
      <Mutation
        mutation={REMOVE_FROM_CATEGORY_MUTATION}
        refetchQueries={[
          {
            query: GAME_SUMMARY_QUERY,
            variables: { gameName },
          },
        ]}
      >
        {(removeFromCategory, { loading, error }) => {
          if (loading) return <p>Removing..</p>;
          if (error) return <Error error={error} />;
          return (
            <Button
              close
              onClick={e => this.handleClick(e, removeFromCategory)}
            >
              <TextLarger>{this.props.children}</TextLarger>
            </Button>
          );
        }}
      </Mutation>
    );
  }
}

export default RemoveFromCategory;
export { REMOVE_FROM_CATEGORY_MUTATION };
