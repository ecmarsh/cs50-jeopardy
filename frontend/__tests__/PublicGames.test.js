import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import wait from 'waait';
import { MockedProvider } from 'react-apollo/test-utils';
import PublicGames from '../components/GamesPublic';
import { fakeGame } from '../lib/testUtils';
import { GAMES_QUERY } from '../components/GamesList';

const { id, name, user } = fakeGame();

const publicGame = {
  id: 'pubGameID',
  name: 'Pubby Game',
  isPublic: true,
  user: { id: 'pgUID' },
};
const privateGame = {
  id,
  name,
  isPublic: false,
  user: { id: user.id },
};

describe('<PublicGames/>', () => {
  it('renders only public games and matches snap', async () => {
    const games = [publicGame, privateGame];
    const gamesMock = {
      request: { query: GAMES_QUERY },
      result: { data: { games } },
    };

    const wrapper = mount(
      <MockedProvider mocks={[gamesMock]} addTypename={false}>
        <PublicGames />
      </MockedProvider>
    );

    expect(wrapper.text()).toContain('Loading');
    await wait(0);
    wrapper.update();

    const form = wrapper.find('form[data-test="public-games-form"]');
    expect(toJSON(form)).toMatchSnapshot();
  });

  it('shows an alert when there are no published games', async () => {
    const games = [privateGame];
    const noPublishedGamesMock = {
      request: { query: GAMES_QUERY },
      result: { data: { games } },
    };
    const wrapper = mount(
      <MockedProvider mocks={[noPublishedGamesMock]} addTypename={false}>
        <PublicGames />
      </MockedProvider>
    );
    await wait(0);
    wrapper.update();

    expect(wrapper.exists('Alert')).toEqual(true);
  });
});
