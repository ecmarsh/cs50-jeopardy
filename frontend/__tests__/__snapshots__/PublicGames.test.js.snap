// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PublicGames/> renders only public games and matches snap 1`] = `
<form
  className="Form-qsq7s5-0 bLtDzK"
  data-test="public-games-form"
  onSubmit={[Function]}
>
  <Utilities__TextEmph>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "Utilities__TextEmph-s11g5i-0",
            "isStatic": false,
            "lastClassName": "jQVwwM",
            "rules": Array [
              "font-weight:bolder;color:",
              [Function],
              ";font-size:1.3em;",
            ],
          },
          "displayName": "Utilities__TextEmph",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "Utilities__TextEmph-s11g5i-0",
          "target": "span",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <span
        className="Utilities__TextEmph-s11g5i-0 jQVwwM"
      >
        Start A Study Game
      </span>
    </StyledComponent>
  </Utilities__TextEmph>
  <fieldset>
    <label
      htmlFor="public-games"
    >
      <Utilities__TextMuted>
        <StyledComponent
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "Utilities__TextMuted-s11g5i-2",
                "isStatic": false,
                "lastClassName": "bdrBqE",
                "rules": Array [
                  "opacity:0.8;font-size:0.8em;display:inline;opacity:1;transition:opacity 0.3s ease-in-out;",
                  [Function],
                  " ",
                  [Function],
                  " ",
                  [Function],
                  " ",
                  [Function],
                ],
              },
              "displayName": "Utilities__TextMuted",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "Utilities__TextMuted-s11g5i-2",
              "target": "span",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
        >
          <span
            className="Utilities__TextMuted-s11g5i-2 bdrBqE"
          >
            Available Games
          </span>
        </StyledComponent>
      </Utilities__TextMuted>
      <select
        autoFocus={true}
        id="public-games"
        name="public-games"
        onChange={[Function]}
        required={true}
        style={
          Object {
            "textTransform": "capitalize",
          }
        }
        value="Select Game"
      >
        <option
          disabled={true}
        >
          Select Game
        </option>
        <option
          key="pubGameID"
          value="Pubby Game"
        >
          Pubby Game
        </option>
      </select>
    </label>
    <Button
      primary={true}
      type="submit"
    >
      <StyledComponent
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "Button-sc-1f1w3gq-0",
              "isStatic": false,
              "lastClassName": "jSaPjy",
              "rules": Array [
                "font-family:",
                [Function],
                ";color:white;border-radius:0.5rem;text-transform:uppercase;letter-spacing:1px;transition:.2s ease;&:hover{cursor:pointer;}&:focus{cursor:pointer;outline:",
                [Function],
                " auto 3px;}",
                [Function],
                " ",
                [Function],
                " ",
                [Function],
                " ",
                [Function],
                " ",
                [Function],
                " &.button--sound{font-weight:bolder;text-decoration:line-through;&::after{content:\\"❌\\";padding-left:3px;}",
                [Function],
                "}",
                [Function],
              ],
            },
            "displayName": "Button",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "Button-sc-1f1w3gq-0",
            "target": "button",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
        primary={true}
        type="submit"
      >
        <button
          className="Button-sc-1f1w3gq-0 jSaPjy"
          type="submit"
        >
          Play Now ▷
        </button>
      </StyledComponent>
    </Button>
  </fieldset>
</form>
`;
